src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -385,12 +467,28 @@ internal string GetDebugInfo(int indent = 0)
         // 3) EscapedNameSection. The escaped verson of NameAsUtf8Bytes plus the wrapping quotes and a trailing colon. Used during serialization.
 
         /// <summary>
-        /// The unescaped name of the property.
-        /// Is either the actual CLR property name,
+        /// The name of the property.
+        /// It's either the actual CLR property name,
         /// the value specified in JsonPropertyNameAttribute,
-        /// or the value returned from PropertyNamingPolicy(clrPropertyName).
+        /// or the value returned from PropertyNamingPolicy.
         /// </summary>
-        internal string Name { get; set; } = null!;
+        public string Name
+        {
+            get => _name;
+            set
+            {
+                CheckMutable();
+
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
+                _name = value;
```

Comment:
Per the existing comments in lines 464-467, shouldn't this also be setting `NameAsUtf8Bytes` and `EscapedNameSection`? Is there a possibility of these falling out of sync or are they computed on initialization?

Reply:
No, we do that as part of `Configure`